src/charm.py

```diff
@@ -93,14 +89,27 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
                 self.unit.status = BlockedStatus("Certificates are not valid")
                 return
             else:
-                self._stored.ca_certificate = self._decode_from_base64_bytes(
-                    base64.b64decode(self.model.config.get("ca-certificate"))  # type: ignore[arg-type]  # noqa: E501
+                replicas = self.model.get_relation("replicas")
```

Comment:
Here both the leader and the non-leader would go through this section and update the peer relation. Is that intended?

Reply:
No, good catch, this would probably crash as non-leaders cannot access the app bucket according to the documentation. I will add a test and fix.

Reply:
Fixed in last commit